iaso/api/projects/serializers.py

```diff
@@ -42,3 +54,17 @@ def validate_app_id(self, data):
     feature_flags = FeatureFlagSerializer(many=True, required=True, allow_empty=True)
     created_at = TimestampField(read_only=True)
     updated_at = TimestampField(read_only=True)
+
+    def to_representation(self, instance):
+        data = super().to_representation(instance)
+        request = self.context.get("request")
+
+        # Only include QR code if the request is paginated
+        if request and (request.query_params.get("page") or request.query_params.get("limit")):
+            qr_image = make_qr_code_image(
+                data='{"url": "' + request.build_absolute_uri("/") + '", "app_id": "' + instance.app_id + '"}',
+                qr_code_options=QRCodeOptions(size="S", image_format="png", error_correction="L"),
+            )
+            data["qr_code"] = f"data:image/png;base64,{base64.b64encode(qr_image).decode('utf-8')}"
+
+        return data
```

Comment:
Would it not be possible to achieve this result with a `SerializerMethodField`?

I don't know what the use cases are for unpaginated calls to the API, but we tend to want to get rid of these, so my instinct would see if I can't use a different serializer for unpaginated requests, for the sake of backwards compatibility.

I would ask @kemar 's opinion here

Reply:
There is something I don't understand. Why do we include the QR code only if the request is paginated?

Reply:
@beygorghor OK, the comment should explain this, otherwise it's almost useless as it's just repeating the code.

Is the computation of the qrcode that slow? If so, `to_representation` is rather appropriate.

Another solution would be to make `qr_code` an "optional" field of the payload, and implement it as a `SerializerMethodField` as suggested by @quang-le